### PR TITLE
Improvements to Player Input component docs

### PR DESCRIPTION
**Pull request by duckets**
### Description

Clarified information about accessing actions via script (ISX-1758) and updated some wording and screenshots to reflect project-wide actions as the default workflow.

https://jira.unity3d.com/browse/ISX-1758

### Changes made

Added notes to Player Input component docs, and Player Input workflow page, and updated screenshots.

### CODE REVIEW

File: Packages/com.unity.inputsystem/Documentation~/PlayerInput.md [hunk 1]
```diff
@@ -12,14 +12,12 @@ The **Player Input** component represents a single player, and that player's ass
...
+*Above, the Player Input component as visible in the inspector.*
 
-Each [`PlayerInput`](../api/UnityEngine.InputSystem.PlayerInput.html) instance represents a separate player or user. You can use multiple `PlayerInput` instances at the same time (although not on the same `GameObject`) to represent local multiplayer setups. The Input System pairs each player to a unique set of Devices that the player uses exclusively, but you can also manually pair Devices in a way that enables two or more players to share a Device (for example, left/right keyboard splits or hot seat use).
+Each [`PlayerInput`](../api/UnityEngine.InputSystem.PlayerInput.html) instance represents a separate player or user. You can use multiple `PlayerInput` instances at the same time (although not on the same `GameObject`) to implement local multiplayer setups. The Input System pairs each player to a unique set of Devices that the player uses exclusively, but you can also manually pair Devices in a way that enables two or more players to share a Device (for example, left/right keyboard splits or hot seat use).
```
Comment (lyndon-unity) [inline Packages/com.unity.inputsystem/Documentation~/PlayerInput.md]: I think you need a PlayerManager Component to do the paring of each player to a unique set of devices. So I think this is wrong when using the PlayerInput component on its own.

File: Packages/com.unity.inputsystem/Documentation~/PlayerInput.md [hunk 2]
```diff
@@ -138,3 +141,9 @@ To set this up, assign a reference to a `InputSystemUIInputModule` component in
...
 
 If you use [`MultiplayerEventSystem`](UISupport.md#multiplayer-uis) components to dispatch UI events, you can also use this setup to simultaneously have multiple UI instances on the screen, each controlled by a separate player.
+
+>**Notes**:
+> - As a general rule, if you are using the PlayerInput workflow, you should read input through callbacks as described above, however if you need to access the input actions asset directly while using the PlayerInput component, you should access the [PlayerInput component's copy of the actions](../api/UnityEngine.InputSystem.PlayerInput.html#UnityEngine_InputSystem_PlayerInput_actions), not `InputSystem.actions`. This is because the PlayerInput component performs device filtering to automatically assign devices to multiple players, so each instance has its own copy of the actions filtered for each player. If you bypass this by reading `InputSystem.actions` directly, the automatic device assignment won't work.
```
Comment (lyndon-unity) [inline Packages/com.unity.inputsystem/Documentation~/PlayerInput.md:160]: Great - thanks for adding

**Review by lyndon-unity — CHANGES_REQUESTED** [review]
Looks good but you need to run the formatting script to fix it before the tests will pass
I use powershell and run .\format.ps1

File: Packages/com.unity.inputsystem/Documentation~/PlayerInput.md (hunk 1 above)
Comment (ekcoh) [reply Packages/com.unity.inputsystem/Documentation~/PlayerInput.md]: You do not need a PlayerManager to achieve the pairing. I believe just need to have multiple PlayerInput component instances. PlayerManager is needed to get the "press to join behavior". Have you observed something different @lyndon-unity ?

File: Packages/com.unity.inputsystem/Documentation~/PlayerInput.md (hunk 2 above)
Comment (ekcoh) [reply Packages/com.unity.inputsystem/Documentation~/PlayerInput.md:160]: Agreed! Thanks for trying to avoid this pitfall via doc guidance

**Review by ekcoh — APPROVED** [review]
LGTM if points by @lyndon-unity are addressed. Hence I will approve since Lyndon already requested changes which blocks the merge until adressed.

File: Packages/com.unity.inputsystem/Documentation~/PlayerInput.md (hunk 1 above)
Comment (lyndon-unity) [reply Packages/com.unity.inputsystem/Documentation~/PlayerInput.md]: Joao tried using player component on its own (multiple instances in the scene) and each responded to all inputs (e.g. gamepad drove all instances). There was no automatic filtering based on which player started the interaction.
I think to get that filtering you have to call some specific functions to choose which devices to filter to. I.e. its not automatic unless using the PlayerInputManager. This definitely confused Joao so I think the above info is needed (or some enhanced variant)

File: Packages/com.unity.inputsystem/Documentation~/PlayerInput.md [hunk 3]
```diff
@@ -1,20 +1,34 @@
...
+
+For example, by using the Player Input component, you can set up a mapping between actions such as "Jump" to C# methods in your script such as `public void OnJump()`.
+
+There are a few options for doing exactly how the Player Input component does this, such as using SendMessage, or Unity Events, which is described in more detail below.
+
+### Handling local multiplayer scenarios
+
+You can also use multiple **Player Input** components (each on a separate instance of a prefab) along with the [**Player Input Manager**](PlayerInputManager.md) component to implement local multiplayer features, such as player lobbies, device filtering, and screen-splitting.
+
+In these local multiplayer scenarios, the Player Input component should be on a prefab which the [**Player Input Manager**](PlayerInputManager.md) has a reference to. The **Player Input Manager** then instantiates players as they join the game and pairs each player instance to a unique device that the player uses exclusively (for example, one gamepad for each player). You can also manually pair devices in a way that enables two or more players to share a Device (for example, left/right keyboard splits or hot seat use).
```
Comment (lyndon-unity) [inline Packages/com.unity.inputsystem/Documentation~/PlayerInput.md:31]: I think we should say something about how it pairs them.
I think its this (but I'd like Hakan's input as I might not be 100% correct)

The system looks up the 'control scheme' associated with the control that is pressed.
If one is found it makes that the only active control scheme applicable for that player.

It also looks at the specific device that the control was pressed on (e.g. The first gamepad connected) and binds that to be the only device associated with the player.

... then we can explain that the WASD could be in control scheme named 'left keyboard' and cursors in a control scheme named 'right keyboard' and two players would be able to play on that same keyboard device.

**Review by lyndon-unity — APPROVED** [review]
Looks good - I've suggest a further enhancement.

**Comment by lyndon-unity** [comment]
Need to run the ./format.ps1 command in powershell

Packages/com.unity.inputsystem/Documentation~/PlayerInput.md
[14:23:05.303 Information] -> 1 file(s) need formatting changes.
...599 total files were examined

**Comment by lyndon-unity** [comment]
Also looks like this page needs a fix

Message: Expected: <empty> But was: < "ActionsAssets.md in C:\build\output\Unity-Technologies\InputSystem\Temp\docstest\com.unity.inputsystem@1.8\manual\PlayerInput.html (File Not Found)" >

**Comment by lyndon-unity** [comment]
> Also looks like this page needs a fix
> 
> Message: Expected: But was: < "ActionsAssets.md in C:\build\output\Unity-Technologies\InputSystem\Temp\docstest\com.unity.inputsystem@1.8\manual\PlayerInput.html (File Not Found)" >

Looks like ActionsAssets should be ActionAssets - drop the s

File: Packages/com.unity.inputsystem/Documentation~/PlayerInput.md (hunk 1 above)
Comment (lyndon-unity) [reply Packages/com.unity.inputsystem/Documentation~/PlayerInput.md]: Ignore this comment as further testing confirmed device binding does work without player input manager.

**Comment by unity-cla-assistant** [comment]
[CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/InputSystem?pullRequest=1827) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/InputSystem?pullRequest=1827) before we can accept your contribution.<br/><hr/>**Lyndon Homewood** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.cds.internal.unity3d.com/check/Unity-Technologies/InputSystem?pullRequest=1827) it.</sub>

**Pull request merged**